conn.go

```diff
@@ -108,10 +113,133 @@ func (c *Conn) RemoteAddr() net.Addr {
 	return c.conn.RemoteAddr()
 }
 
-// Write takes a frisbee.Message and some (optional) accompanying content, and queues it up to send asynchronously.
+// Write takes a byte slice and sends a BUFFER message
+func (c *Conn) Write(p []byte) (int, error) {
+
+	var encodedMessage [protocol.MessageV0Size]byte
+
+	binary.BigEndian.PutUint16(encodedMessage[protocol.VersionV0Offset:protocol.VersionV0Offset+protocol.VersionV0Size], protocol.Version0)
+	binary.BigEndian.PutUint32(encodedMessage[protocol.OperationV0Offset:protocol.OperationV0Offset+protocol.OperationV0Size], BUFFER)
+	binary.BigEndian.PutUint64(encodedMessage[protocol.ContentLengthV0Offset:protocol.ContentLengthV0Offset+protocol.ContentLengthV0Size], uint64(len(p)))
+
+	c.Lock()
+	if c.state.Load() != CONNECTED {
+		c.Unlock()
+		return 0, c.Error()
+	}
+
+	_, err := c.writer.Write(encodedMessage[:])
+	if err != nil {
+		c.Unlock()
+		if c.state.Load() != CONNECTED {
+			err = c.Error()
+			c.logger.Error().Msgf(errors.WithContext(err, WRITE).Error())
+			return 0, errors.WithContext(err, WRITE)
+		}
+		c.logger.Error().Msgf(errors.WithContext(err, WRITE).Error())
+		return 0, c.closeWithError(err)
+	}
+
+	_, err = c.writer.Write(p)
+	if err != nil {
+		c.Unlock()
+		if c.state.Load() != CONNECTED {
+			err = c.Error()
+			c.logger.Error().Msgf(errors.WithContext(err, WRITE).Error())
+			return 0, errors.WithContext(err, WRITE)
+		}
+		c.logger.Error().Msgf(errors.WithContext(err, WRITE).Error())
+		return 0, c.closeWithError(err)
+	}
+
+	if len(c.flusher) == 0 {
+		select {
+		case c.flusher <- struct{}{}:
+		default:
+		}
+	}
+
+	c.Unlock()
+
+	return len(p), nil
+}
+
+// ReadFrom is a function that will send buffer messages from an io.Reader until EOF or an error occurs
+// In the event that the connection is closed, ReadFrom will return an error.
+func (c *Conn) ReadFrom(r io.Reader) (n int64, err error) {
+	buf := make([]byte, 1<<19)
+
+	var encodedMessage [protocol.MessageV0Size]byte
+
+	binary.BigEndian.PutUint16(encodedMessage[protocol.VersionV0Offset:protocol.VersionV0Offset+protocol.VersionV0Size], protocol.Version0)
+	binary.BigEndian.PutUint32(encodedMessage[protocol.OperationV0Offset:protocol.OperationV0Offset+protocol.OperationV0Size], BUFFER)
+
+	for err == nil {
```

Comment:
I dont think err will ever be non-nil since we break or return on every err case. Thats fine but this should just be an infinite for explicitly